src/libawkward/array/UnionArray.cpp

```diff
@@ -270,6 +270,15 @@ namespace awkward {
     return out;
   }
 
+  bool
+  UnionForm::istuple() const {
+  bool is_tuple = false;
+    for (auto content : contents_) {
+        is_tuple = is_tuple && content.get()->istuple();
+    }
+    return is_tuple;
+  }
```

Comment:
I just noticed this, not sure if it's in v2 also, but how could the result of this ever be true? `false && X` is `false`.

Reply:
@jpivarski Yep, that's wrong (I inverted the condition). I should add a test for the union too.